src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -185,6 +185,10 @@ protected function validateType(&$value, $type, $coerce = false)
         }
 
         if ('array' === $type) {
+            if ($coerce) {
+                $value = $this->toArray($value);
```

Comment:
Hmm... will have to think on this a bit. At a minimum I should write some tests to check that post-coercion validation is happening properly.